raft/fuzz_test.go

```diff
@@ -0,0 +1,215 @@
+// Copyright 2022 The etcd Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package raft
+
+import (
+	"os"
+	"runtime"
+	"strings"
+	"sync"
+	"testing"
+
+	fuzz "github.com/AdaLogics/go-fuzz-headers"
+	pb "go.etcd.io/etcd/raft/v3/raftpb"
+
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
```

Comment:
Minor comment: let's try to get third-party dependencies included in one group, and etcd dependencies in another group.
```suggestion
	fuzz "github.com/AdaLogics/go-fuzz-headers"
	"go.uber.org/zap"
	"go.uber.org/zap/zapcore"
	
	pb "go.etcd.io/etcd/raft/v3/raftpb"
```